<!-- Create the launch file for ROS -->
<launch>
	<node name="gen1" pkg="simulator" type="generator.py" output="screen">
		<param name="node_id" value="1"/>
		<param name="node_name" value="gen1"/>
		<param name="next_element" value="switch"/>
		<param name="gen_freq" value="10"/>
		<param name="num_messages" value="5"/>
		<param name="event_type" value="test_typw"/>
		<param name="attribute1" value="test"/>
		<param name="value1" value="0.3"/>
	</node>

	<node name="coda1" pkg="simulator" type="coda.py" output="screen">
		<param name="node_id" value="1"/>
		<param name="node_name" value="coda1"/>
		<param name="next_element" value="end_node"/>
		<param name="num_servers" value="3"/>
		<param name="server_time" value="0.3"/>
	</node>

	<node name="coda2" pkg="simulator" type="coda.py" output="screen">
		<param name="node_id" value="2"/>
		<param name="node_name" value="coda2"/>
		<param name="next_element" value="end_node"/>
		<param name="num_servers" value="1"/>
		<param name="server_time" value="0.2"/>
	</node>

	<node name="switch" pkg="simulator" type="switch2.py" output="screen">
		<param name="node_id" value="1"/>
		<param name="node_name" value="switch"/>
		<param name="modality" value="split_attribute"/>
		<param name="split_attribute" value="test"/>
		<param name="topic1" value="coda1"/>
		<param name="topic2" value="coda2"/>
		<param name="attribute_num" value="1"/>
	</node>

	<node name="end_node" pkg="simulator" type="end_node.py" output="screen">
		<param name="node_id" value="4"/>
		<param name="node_name" value="end_node"/>
		<param name="modality" value="events"/>
	</node>

</launch>

